### PR TITLE
citgm-stress: add a citgm stress test

### PR DESCRIPTION
**Pull request by gdams**
Closes: #122
The idea behind citgm stress is that we can repeat a single module test
multiple times to test for flakiness. You can run citgm-stress using
`./bin/citgm-stress.js <module> <repeat>`

##### Checklist

- [x] `npm test` passes
- [x] tests are included
- [x] documentation is changed or added
- [x] contribution guidelines followed [here](https://github.com/nodejs/citgm/blob/master/CONTRIBUTING.md)

#### TODO
- [ ] write description for README.md

### CODE REVIEW

**Comment by gdams** [comment]
So this is the first fully working form of the citgm stress job that I have built. There is still a fair amount of work left and I'm sure everyone will have suggestions

**Comment by MylesBorins** [comment]
@gdams is this a WIP? if so we should add it to the title... or maybe make a label?

**Comment by gibfahn** [comment]
>@gdams is this a WIP? if so we should add it to the title... or maybe make a label?
 
+1 to both

**Comment by gdams** [comment]
@nodejs/citgm can I get some feedback about this now? I have made the tests so it's just of case of whether everyone is happy with the way that citgm-stress works

File: README.md [hunk 1]
```diff
@@ -141,6 +141,36 @@ For syntax, see [lookup.json](./lib/lookup.json), the available attributes are:
...
+  -h, --help                  output usage information
+  -V, --version               output the version number
+  --config                    Path to a JSON config file
+  -v, --verbose [level]       Verbose output (silly, verbose, info, warn, error)
+  -q, --npm-loglevel [level]  Verbose output (silent, error, warn, http, info, verbose, silly)
+  -l, --lookup <path>         Use the lookup table provided at <path>
+  -d, --nodedir <path>        Path to the node source to use when compiling native addons
+  -p, --test-path <path>      Path to prepend to $PATH when running tests
+  -n, --no-color              Turns off colorized output
+  -s, --su                    Allow running the tool as root.
```
Comment (richardlau) [inline README.md]: The description here ends with a full stop but the others do not.

File: README.md [hunk 2]
```diff
@@ -141,6 +141,36 @@ For syntax, see [lookup.json](./lib/lookup.json), the available attributes are:
 If you want to pass options to npm, eg `--registry`, you can usually define an
 environment variable, eg `"npm_config_registry": "https://www.xyz.com"`.
 
+## citgm-stress
+
+//Need to write a description here//
```
Comment (richardlau) [inline README.md]: :grin:

File: bin/citgm-stress.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  .usage('citgm-stress [options] <module> <repeat>')
+  .option('parallel', {
+    alias: 'j',
+    type: 'number',
+    description: 'Number of tests to run in parallel'
+  })
+  .option('autoParallel', {
+    alias: 'J',
+    type: 'boolean',
+    description: 'Auto detect number of cores to use to run tests in parallel'
```
Comment (richardlau) [inline bin/citgm-stress.js]: These options aren't documented in the readme or man page?

File: README.md (hunk 2 above)
Comment (gdams) [inline README.md]: That is in the TODO list above, I wanted to get everyone's opinion on a suitable description CC @nodejs/citgm

File: README.md (hunk 1 above)
Comment (gdams) [inline README.md]: removed

File: bin/citgm-stress.js (hunk 3 above)
Comment (gdams) [inline bin/citgm-stress.js]: updated

**Comment by gdams** [comment]
@nodejs/citgm could I get some reviews please

**Comment by gibfahn** [comment]
I don't understand why we have to add an entirely new file to have a `repeat` option. I don't like to say "I don't think this is the right way to do this" unless I can back it up with code, so I had a go in https://github.com/nodejs/citgm/pull/411.

**Comment by MylesBorins** [comment]
@gibfahn doesn't this make sense as a feature for citgm rather than citgm-all? Would we want to be running multiple iterations for all?

**Comment by gibfahn** [comment]
>@gibfahn doesn't this make sense as a feature for citgm rather than citgm-all? Would we want to be running multiple iterations for all?

I think it makes sense as a feature for both. What I don't get is the need to have a separate command for what can just be an option you pass to the script.

**Comment by MylesBorins** [comment]
@gibfahn are you still working on this or should we close

**Comment by MylesBorins** [comment]
Closing for now, feel free to reopen

**Pull request closed without merging**